uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/api/v3/LeaveRequestTest.php

```diff
@@ -3898,6 +3898,147 @@ public function testGetAndGetFullReturnsAllDataWhenLoggedInUserHasEditAllContact
     $this->assertEquals($contact2['id'], $result['values'][1]['contact_id']);
   }
 
+  public function testGetAndGetFullIncludesAllTheRequestsUpToTheLastDayMinuteIfToDateIsPresentAndHourIsEmpty() {
+    $contract1 = HRJobContractFabricator::fabricate(
+      [ 'contact_id' => 1 ],
+      [ 'period_start_date' => '2016-01-01' ]
+    );
+
+    $contract2 = HRJobContractFabricator::fabricate(
+      [ 'contact_id' => 2 ],
+      [ 'period_start_date' => '2016-01-01' ]
+    );
+
+    LeaveRequestFabricator::fabricateWithoutValidation([
+      'contact_id' => $contract1['contact_id'],
+      'type_id' => 1,
+      'from_date' => CRM_Utils_Date::processDate('2016-02-20'), //Hour will be automatically set to 00:00:00
+      'to_date' =>  CRM_Utils_Date::processDate('2016-02-23'), //Hour will be automatically set to 23:59:59
+    ], true);
+
+    LeaveRequestFabricator::fabricateWithoutValidation([
+      'contact_id' => $contract2['contact_id'],
+      'type_id' => 1,
+      'from_date' => CRM_Utils_Date::processDate('2016-02-23 10:00:00'),
+      'to_date' =>  CRM_Utils_Date::processDate('2016-02-23 16:00:00'),
+    ], true);
+
+    $result = civicrm_api3('LeaveRequest', 'get', [
+      'to_date' => ['<=' => '2016-02-23'],
+      'sequential' => 1
+    ]);
+    $this->assertEquals(2, $result['count']);
+    $this->assertEquals($contract1['contact_id'], $result['values'][0]['contact_id']);
+    $this->assertEquals($contract2['contact_id'], $result['values'][1]['contact_id']);
+
+    $result = civicrm_api3('LeaveRequest', 'getFull', [
+      'to_date' => ['<=' => '2016-02-23'],
+      'sequential' => 1
+    ]);
+    $this->assertEquals(2, $result['count']);
+    $this->assertEquals($contract1['contact_id'], $result['values'][0]['contact_id']);
+    $this->assertEquals($contract2['contact_id'], $result['values'][1]['contact_id']);
+  }
+
+  public function testGetAndGetFullIncludesAllTheRequestsStartingFromTheFirstMinuteOfTheDayMinuteIfFromDateIsPresentAndHourIsEmpty() {
+    $contract1 = HRJobContractFabricator::fabricate(
+      [ 'contact_id' => 1 ],
+      [ 'period_start_date' => '2016-01-01' ]
+    );
+
+    $contract2 = HRJobContractFabricator::fabricate(
+      [ 'contact_id' => 2 ],
+      [ 'period_start_date' => '2016-01-01' ]
+    );
+
+    LeaveRequestFabricator::fabricateWithoutValidation([
+      'contact_id' => $contract1['contact_id'],
+      'type_id' => 1,
+      'from_date' => CRM_Utils_Date::processDate('2016-02-20'), //Hour will be automatically set to 00:00:00
+      'to_date' =>  CRM_Utils_Date::processDate('2016-02-20'), //Hour will be automatically set to 23:59:59
+    ], true);
+
+    LeaveRequestFabricator::fabricateWithoutValidation([
+      'contact_id' => $contract2['contact_id'],
+      'type_id' => 1,
+      'from_date' => CRM_Utils_Date::processDate('2016-02-20 23:59:59'),
+      'to_date' =>  CRM_Utils_Date::processDate('2016-02-23 16:00:00'), //Hour will be automatically set to 23:59:59
```

Comment:
Why will the hour automatically be changed in `to_date`?

Reply:
It won't. This is a copy and paste mistake. Good catch